scipy-stubs/sparse/_construct.pyi

```diff
@@ -74,7 +77,9 @@ _FmtDIA: TypeAlias = Literal["dia"]
 _FmtDOK: TypeAlias = Literal["dok"]
 _FmtLIL: TypeAlias = Literal["lil"]
 _FmtNonCOO: TypeAlias = Literal["bsr", "csc", "csr", "dia", "dok", "lil"]
-_FmtNonDIA: TypeAlias = Literal["bsr", "coo", "csc", "csr", "dok", "lil"]
+
+_Diag: TypeAlias = _ToArray1D[_SCT] | _ToArray2D[_SCT]
```

Comment:
Hmm, I'm not sure about this name; it's not very descriptive. So how about something like `_ToArray1D2D` or ``_ToArrayMax2D` or something?